content/technical/open-source-as-a-service.md

```diff
@@ -0,0 +1,153 @@
+common place.
+
+And yet there's a very simple principle that drives Open Source:
+
+<div style="font-weight: bold; text-align: center;">
+Open Source thrives when it converts some<br>small percentage of users into contributors.
+</div>
+
+If you interrupt that principle, you starve Open Source. Software as a
+Service does just that: When someone else runs software for you, the
+intuitive mechanisms for users to change that software are unavailable.
+
+"But" you may say, "If the source code for that running service was **open**
+then I could still change it."
+
+Sure you could change the code in some component, but for any reasonably
+complex Service you wouldn't be able to run your changes, experience your
+changes against real world workflows, much less iterate on them until
+they're good enough to share with others.
+
+In reality the threat here is far more fundamental: Users of services
+cannot readily change something in the service not only because the
+processes is operated by someone else ... but because such users of
+a service have explicitly chosen not to be involved in operating
+the software.
+
+If the primary way to use [Postgres was as a Service](https://aws.amazon.com/rds/),
+then that (fabulous) project would be starved for contributors. This is
+because the number of contributors in a such a project are a function
+of some small percentage of users deciding to try to make change.
+
+As this mechanism for using software, as a service, takes over the world
+the pool of users who can contribute to Open Source shrinks dramatically.
+
+For a long time I saw this as a fundamental threat that would starve
+Open Source. I was unable reconcile Software as a Service with a healthy
+Open Source ecosystem.
+
+But recently, I've become convinced that if we enable an effective
+contribution model on a service, one that doesn't require that the users
+themselves become operators of the service ... then we can reconcile this
+threat and Open Source will thrive on services instead of be starved by them.
+
+
+# You want me to do what!?
+
+To enable contributions by users of a service, users that have explicitly
+chosen not to operate the software themselves, we have to set up a process
+by which they can make a change to a running service and experience that
+change themselves before others do.
+
+For example, a user of such a hypothetical service should be able to:
+
+ 1. Discover which component of a service to contribute to.
+
+ 2. Make a non-sensical change (like add a printf style log statement) or change the spelling of a word.
+
+ 3. Experience that change when they use the service, or when it acts on their data.
+
+After discussing this with others, I firmly believe we have all the techniques we need,
+whether canary deployments, load balancing, continuous delivery,
+infrastructure as code, and so on. It's been hard to find a single
+problem that an existing practice doesn't solve.
```

Comment:
Not for your article (perhaps), but a thought: Achieving this would also mean that you have full control of the "real" deployment, and constantly exercise and improve its (re)deployment. The importance of that can hardly be over-estimated for a project's/company's long-term resilience.